src/gpl/src/graphics.cpp

```diff
@@ -508,4 +508,57 @@ bool Graphics::guiActive()
   return gui::Gui::enabled();
 }
 
+void Graphics::saveGuiImage(const std::string& filename)
+{
+  if (gui::Gui::enabled()) {
+    gui::Gui::get()->saveImage(filename.c_str());
+  } else {
+    logger_->report("GUI is not active. Cannot save '{}'.", filename);
+  }
+}
+
+void Graphics::scaleAndAnnotateImage(const std::string& input_path,
+                                     const std::string& output_path,
+                                     const std::string& label,
+                                     const std::string& fill_color)
+{
+  std::string cmd = fmt::format(
+      "convert {} -resize 50% -colors 64 -strip -quality 85 "
+      "-gravity SouthEast -pointsize 20 -fill {} "
+      "-annotate +5+5 '{}' PNG8:{}",
+      input_path,
+      fill_color,
+      label,
+      output_path);
+
+  int ret = std::system(cmd);
```

Comment:
not a huge fan of calls to `system`, seems like the GUI could be augmented to support both the image and gif generation. Separate from this PR, but might be a nice thing to support anyways.

Reply:
Perhaps https://github.com/charlietangora/gif-h - single header library that we could put in third-party